app/admin.rb

```diff
@@ -3,24 +3,37 @@
 module Admin
   extend Discordrb::Commands::CommandContainer
 
-  command :adddeath, help_available: false do |event, *parse_string|
+  def self.find_role(event, role)
     role = nil
-    parse_string = parse_string.join(' ')
-    parse_string = parse_string.split('-', 2)
-    name = parse_string[0].chomp
-    reason = parse_string[1].chomp.gsub!(/^\s+/, '')
-
-    puts name
-    puts reason
-
     event.server.roles.each do |server_role|
       next unless server_role.name == 'botadmin'
       role = server_role
-      break
     end
+    return role
+  end
 
+  command :updateprofile, help_available: false do |event|
+    role = find_role(event, 'botadmin')
     break unless event.user.role?(role)
```

Comment:
Yes, but that means you have to program a way to add bot permissions to the bot, and persistently store that. I'd rather offload it to a botadmin role for now, and refactor later, after we actually have a way to verify who someone is. At least this way Discord already has done that.

Also, you're right, I could do a method call in a single line. I was half-tempted to do it that way originally. I'll change it. :)